### DISCUSSION TITLE
Is the rule allowing only one group in bounding object really necessary/a good thing?

### DISCUSSION THREAD

**Discussion by BenjaminDeleze** [cyberbotics/webots#6132] [General]
I am trying to build a bounding object for the following object:
![test2](https://github.com/cyberbotics/webots/assets/25938827/ac302f32-f876-41b9-b305-cb2d7f00c1ba)

The idea is:

- Two capsule for the straight parts
- A slighty modified `TorusBoundingObject` for the round part.

However I am not able to have:

```
boundingObject Group {
  DEF CAPSULE1 Transform {...}
  DEF CAPSULE2 Transform {...}
  HalfTorusBoundingObject {...}
}
```
because the torus bounding object is a group of Sphere.
So for it to work, I currently have to:
- insert the Torus bounding object
- convert it to base node
- add the capsules in it

Which is not convenient as I am experimenting different sizes of torus. (So every time I have to delete everything and start again instead of just changing the parameter of the torus and then adjusting the position of the capsule).

**Comment by omichel** [comment]
Why don't you create a PROTO of LassoBoundingObject with a size parameter that generates both the capsules and the half torus as spheres?

**Reply by BenjaminDeleze** [reply]
Because it seems overly complicated for what I want to achieve.
It could be something that I will do once I found something I am pleased with. But for now I am testing different angles, size, radius, number of capsule,... and it is way easier to do it in the scene tree and with the handles.

**Comment by omichel** [comment]
But I agree, we could relax this constraint and allow any Group/Pose combination which we could automatically convert in Webots into a single Group containing Poses.
